backend/src/controllers/BuildingController.js

```diff
@@ -0,0 +1,33 @@
+export class BuildingController {
+
+    constructor({ResourceService,BuildingService,ResourceRepo,BuildingRepo,db,errorCodes}) {
+        this.building = new BuildingService({BuildingRepo,ResourceService,ResourceRepo,db,errorCodes});
+        this.post = this.post.bind(this);
+        this.errorMessages = {
+            [errorCodes.missingKingdomId]: {status: 400, message: 'KingdomId is missing.'},
+            [errorCodes.missingBuildingType]: {status: 400, message: 'Building type is required.'},
+            [errorCodes.missingResourceType]: {status: 400, message: 'Missing resource type.'},
+            [errorCodes.missingResourceAmount]: {status: 400, message: 'Missing resource amount.'},
+            [errorCodes.missingResourceGeneration]: {status: 400, message: 'Missing resource generation.'},
+            [errorCodes.invalidKingdomId]: {status: 400, message: 'Invalid kingdomId.'},
+            [errorCodes.invalidBuildingType]: {status: 400, message: 'Wrong building type.'},
+            [errorCodes.invalidResourceAmount]: {status: 400, message: 'Not enough gold.'},
+        };
```

Comment:
@gulyaasjaanos these doesn't have to be a class property, can be just a const outside of the class in this file, but it's fine like this